src/job-exporter/src/collector.py

```diff
@@ -267,13 +294,76 @@ class GpuCollector(Collector):
     cmd_histogram = Histogram("cmd_nvidia_smi_latency_seconds",
             "Command call latency for nvidia-smi (seconds)")
 
-    cmd_timeout = 3 # 99th latency is 0.97s
+    cmd_timeout = 60 # 99th latency is 0.97s
 
-    def __init__(self, name, sleep_time, atomic_ref, iteration_counter, gpu_info_ref):
+    def __init__(self, name, sleep_time, atomic_ref, iteration_counter, gpu_info_ref, zombie_info_ref):
         Collector.__init__(self, name, sleep_time, atomic_ref, iteration_counter)
         self.gpu_info_ref = gpu_info_ref
+        self.zombie_info_ref = zombie_info_ref
+
+    @staticmethod
+    def get_container_id(pid):
+        """ return two values, the first one is if we found the corresponding
+        container_id, the second one is the container_id if found """
+        path = "/proc/%d/cgroup" % (pid)
+        if not os.path.isfile(path):
+            return False, ""
+
+        with open(path) as f:
+            content = f.read()
+
+        for line in content.split("\n"):
+            line = line.strip()
+            if "pids" in line and "/docker/" in line:
+                parts = line.split("/docker/")
```

Comment:
Could we also consider kubernetes pod? Then if we switch to static pod, we won't need to change here.

Reply:
How to consider k8s pod?

Reply:
@xudifsd 
Not find a clean definition of k8s pod cgroup, we could add a TODO here to remind us.